Code/Framework/AzToolsFramework/Tests/ComponentModeSwitcherTests.cpp

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ComponentModeTestDoubles.h"
+#include "ComponentModeTestFixture.h"
+
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <AzFramework/Viewport/ViewportScreen.h>
+#include <AzTest/AzTest.h>
+#include <AzToolsFramework/ViewportUi/ButtonGroup.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiSwitcher.h>
+#include <AzToolsFramework/ComponentMode/ComponentModeSwitcher.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiManager.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/Application/ToolsApplication.h>
+#include <AzToolsFramework/API/EntityCompositionRequestBus.h>
+
+namespace UnitTest
+{
+    using namespace AzToolsFramework;
+    using namespace AzToolsFramework::ComponentModeFramework;
+
+    using ComponentModeSwitcher = AzToolsFramework::ComponentModeFramework::ComponentModeSwitcher;
+
+    class ComponentModeSwitcherTestFixture : public ComponentModeTestFixture
+    {
+    };
+
+    TEST_F(ComponentModeSwitcherTestFixture, AddComponentToEntityAddsComponentToSwitcher)
+    {
+        // Given
```

Comment:
update: I think these tests look great but to help the reader understand them more quickly, have a brief description under each `Given/When/Then` comment would be really nice to indicate what is happening at a high level and what is being checked (what the code maps back to in Editor concepts if that makes sense)